django/views/generic/detail.py

```diff
@@ -64,13 +73,13 @@ def get_queryset(self):
         `get_object` and may not be called if `get_object` is overridden.
         """
         if self.queryset is None:
-            if self.model:
-                return self.model._default_manager.all()
+            if self.get_model():
+                return self.get_model()._default_manager.all()
             else:
                 raise ImproperlyConfigured(
                     "%(cls)s is missing a QuerySet. Define "
-                    "%(cls)s.model, %(cls)s.queryset, or override "
-                    "%(cls)s.get_queryset()." % {
+                    "%(cls)s.model, %(cls)s.queryset, a forms.ModelForm as %(cls)s.form_class, "
+                    "or override %(cls)s.get_queryset()." % {
```

Comment:
I think `SingleObjectMixin` should be completely agnostic about what `ModelFormMixin`. What about suggesting to override `get_model()` instead like we do with `get_queryset()`?